### ISSUE TITLE
Listening for Metadata loaded event

### ISSUE THREAD

**Issue by josephharding** [clappr/clappr#501]
If I want to know when my HLS stream has completed loading metadata, how can I listen for that event?

I've tried something like this, but with no luck:

`player.on(Clappr.Events.PLAYBACK_LOADEDMETADATA, function() { console.log("metadata loaded"); });`

Thanks!

**Comment by flavioribeiro** [comment]
hey @josephharding just for curiosity, what do u mean by metadata? Since clappr uses [flashls](flashls.org) for HLS playback, I'm pretty sure it should already support (or support soon when the guys update the swf) all events described [here](https://github.com/mangui/flashls/blob/dev/API.md).

**Comment by josephharding** [comment]
@flavioribeiro - thanks for responding so quickly!

Specifically I'm looking for the loading event at which the duration of the video would be available.

In the HTML5 video player you can listen for that event like this:

`videoElement.addEventListener("loadedmetadata", this.onLoadedMetaData.bind(this, videoElement));`

at which point `videoElement.duration` is populated.

Looking through the Clappr source I saw this guy:

Events.PLAYBACK_LOADEDMETADATA

And just assumed that clappr would wrap the flash api and fire this event - I hadn't tried listening directly to the flash player.

**Comment by towerz** [comment]
@josephharding We've just released a new version (0.1.7) with this event implemented in the HLS playback. It's not exposed directly by the player, but you can use the container event:
```javascript
player.core.getCurrentContainer().on(Clappr.Events.CONTAINER_LOADEDMETADATA, function(duration) { console.log('loaded video with duration', duration, 'seconds') }
```
Please let me know if this will work for you.

**Comment by josephharding** [comment]
@towerz - just tried it out, works perfectly, thanks!
